crates/common/src/ckb_decimal.rs

```diff
@@ -0,0 +1,34 @@
+use gw_types::U256;
+
+// NOTE: u64::MAX is about 10^18, U256::MAX is about 10^77. Multiple 10 pow
+// `CKB_DECIMAL_POW_EXP` should not overflow.
+pub const CKB_DECIMAL_POW_EXP: u32 = 10;
+pub const CKB_DECIMAL_POWER_TEN: u64 = 10u64.pow(CKB_DECIMAL_POW_EXP);
+
+pub fn to_18(amount: u64) -> U256 {
+    U256::from(amount) * CKB_DECIMAL_POWER_TEN
+}
+
+pub fn from_18(amount: U256) -> Option<u64> {
+    let truncated = amount / CKB_DECIMAL_POWER_TEN;
+    if truncated.bits() > u64::BITS as usize {
+        None
+    } else {
+        Some(truncated.as_u64())
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
It's good to be cautious, but overflows in constant expressions can be caught at compile time, so the test is a bit redundant.